File: C#/asymptote.cs
using System;

public class Asymptote {
    public static void Main(string[] args) {
        double p, n, u;

        Console.WriteLine("\"p\" value: ");
        p = Convert.ToInt32(Console.ReadLine());
        Console.WriteLine("\"n\" value: ");
        n = Convert.ToInt32(Console.ReadLine());
        u = (2 * n + 3) / (n + 1);

        while (Math.Abs(u - 2) >= Math.Pow(10, -p))
        {
            n++;
            u = (2 * n + 3) / (n + 1);
        }

        Console.WriteLine(n);
    }
}

File: PHP/asymptote.php
<?php
$p = readline("\"p\" value: ");
$n = 0;
$u = 3;

while (abs($u - 2) >= pow(10, -$p)) {
    $n += 1;
    $u = (2*$n +3)/($n + 1);
}

echo($n);
?>
